toolkit/scripts/check_signatures.py

```diff
@@ -20,20 +21,75 @@ def getSignature(fileName) -> str:
             sha256sum.update(read_data)
     return sha256sum.hexdigest()
 
-def check_folder(path):
+def find_file_and_check(path, filename, expected_signature) -> Optional[bool]:
+    path_to_check = os.path.join(path, filename)
+    if Path(path_to_check).is_file():
+        actual_signature = getSignature(path_to_check)
+        if actual_signature == expected_signature:
+            return True
+        print(f"ERROR: detected a mismatched signature for {filename}, expected [{expected_signature}] does not equal actual [{actual_signature}]")
+        return False
+
+    for content in os.listdir(path):
+        path_to_check = os.path.join(path, content)
+        if os.path.isdir(path_to_check):
+            result = find_file_and_check(path_to_check, filename, expected_signature)
```

Comment:
It feels like we're re-implementing a recursive file search. What do you think about using [Python modules](https://stackoverflow.com/questions/2186525/how-to-use-to-find-files-recursively), that would do it for us?

Reply:
maybe, but the time it took me to find something vs just writing this didn't feel justified.

Reply:
replaced this code with Path.glob.